ql/src/java/org/apache/hadoop/hive/ql/io/orc/OrcSplit.java

```diff
@@ -104,7 +103,15 @@ public OrcSplit(Path path, Object fileId, long offset, long length, String[] hos
     this.isOriginal = isOriginal;
     this.hasBase = hasBase;
     this.rootDir = rootDir;
-    this.deltas.addAll(filterDeltasByBucketId(deltas, AcidUtils.parseBucketId(path)));
+    int bucketId = AcidUtils.parseBucketId(path);
```

Comment:
@difin @deniskuzZ 
The local variable hides the instance variable; probably is was intentional but it is quite confusing; especially for someone that's gonna debug this code in the future. A comment/explanation would definitely help.

Reply:
@zabetak nice catch, I didn't notice that there is an instance variable`bucketId` cause it wasn't initialized in a constructor, but in a parse() method. The whole thing needs to be refactored.

Reply:
Hi @deniskuzZ @zabetak 
Fixed in PR https://github.com/apache/hive/pull/3606